### ISSUE TITLE
constructing Node from NaN and Inf emits "nan" and "inf" respectively, breaking the core schema regexes.

### ISSUE THREAD

**Issue by leni536** [jbeder/yaml-cpp#507]
```cpp
#include <iostream>
#include <limits>

#include <yaml-cpp/yaml.h>

int main() {
	auto yaml_node = YAML::Node( std::numeric_limits<double>::infinity() );
	std::cout << yaml_node << std::endl;
	return 0;
}
```

Output: *inf"
Expected output: ".inf", "+.inf", ".Inf" ... see [YAML 1.2 Core schema tag resolution](http://yaml.org/spec/1.2/spec.html#id2805071).

Notable that yaml-cpp *correctly* throws YAML::TypedBadConversion<double> if one tries to parse "inf" as double.

**Comment by jkhoogland** [comment]
is it possible to make the pull request pass the tests and merge it ?

**Comment by leni536** [comment]
I take a look at it.

**Comment by leni536** [comment]
I think this specific issue is fixed in e0e01d53c27ffee6c86153fa41e7f5e57d3e5c90.

**Comment by jkhoogland** [comment]
ah, great! thanks
